dace/codegen/targets/intel_fpga.py

```diff
@@ -524,26 +542,51 @@ def generate_kernel_internal(self, sdfg, state, kernel_name, subgraphs,
         kernel_stream.write(kernel_header_stream.getvalue() +
                             kernel_body_stream.getvalue())
 
-        self.generate_host_function_epilogue(sdfg, state, host_code_body_stream)
+        # Generate host kernel invocation
+        self.generate_host_function_body(sdfg, state, state_host_body_stream,
+                                         kernel_name, predecessors)
 
-        # Store code to be passed to compilation phase
-        self._host_codes.append(
-            (kernel_name, host_code_header_stream.getvalue() +
-             host_code_body_stream.getvalue()))
+    def generate_host_function_prologue(self, sdfg, state, host_stream,
+                                        kernel_name):
+        seperator = "/" * 59
+        host_stream.write(
+            f"\n{seperator}\n// Kernel: {kernel_name}\n{seperator}\n\n")
 
-    @staticmethod
-    def generate_host_function_prologue(sdfg, state, host_stream):
-        host_stream.write("std::vector<hlslib::ocl::Kernel> kernels;", sdfg,
-                          sdfg.node_id(state))
+        host_stream.write(
+            f"std::vector<hlslib::ocl::Kernel> {kernel_name}_kernels;", sdfg,
```

Comment:
Can we just make this a constant-sized array instead? We don't need to dynamic capabilities of `std::vector`

Reply:
This would require assigning an increasing ID to all the modules (PEs), since the code for populating that vector is in `generate_module` (vendor specific, called by `fpga.py`)

Reply:
Aren't they ordered anyway?